syntax = "proto3";

package mutex;

service Poll {
	rpc poll (PollRequest) returns (PollReply) {}
}

service RecvRequest {
	rpc recvRequest (RecvReq) returns (RecvReply) {}
}

service SendDefReply {
	rpc sendDefRep (DefRep) returns (DefReply) {}
}

message PollRequest {

}

message PollReply {
	bool active = 1;
}

message RecvReq {

}

message RecvReply {
	bool reply = 1;
}

message DefRep {

}

message DefReply {
	bool defer = 1;
}
